### PR TITLE
copying file path and directory to clipboard

### PR DESCRIPTION
**Pull request by makeefu**
Some simple ranger functionality concerning copying file names, file paths and current directories to clipboard added.

- New public functions `copy_filepath` and `copy_dir` which behave similarly to `copy_filename`.

- New `KeyCommand`s `CopyFilePath` and `CopyDir` behave like `CopyFileName`.

- Now one can copy file name, file path and current directory by calling `copy_filename`, `copy_filepath` and `copy_dir` commands.

- Respective shortcuts added to keymap.toml

### CODE REVIEW

File: src/commands/file_ops.rs [hunk 1]
```diff
@@ -49,45 +49,92 @@ pub fn paste(context: &mut AppContext, options: IoWorkerOptions) -> JoshutoResul
...
+        .and_then(|entry| entry.file_path().to_str())
+        .and_then(|s| Some(s.to_string()));
+
+    if let Some(file_path) = entry_file_path {
+        copy_string_to_buffer(file_path)?;
+    }
+    Ok(())
+}
+
+pub fn copy_dir(context: &mut AppContext) -> JoshutoResult<()> {
```
Comment (kamiyaa) [inline src/commands/file_ops.rs:80]: Could you change this to `copy_dirname` and same goes for `CopyDir` -> `CopyDirName`, just so its less ambiguous.

File: src/commands/file_ops.rs [hunk 2]
```diff
@@ -49,45 +49,92 @@ pub fn paste(context: &mut AppContext, options: IoWorkerOptions) -> JoshutoResul
...
+    let opt_entry = context
+        .tab_context_ref()
+        .curr_tab_ref()
+        .curr_list_ref()
+        .and_then(|c| c.curr_entry_ref());
+
+    let dir:  Option<String> = opt_entry
```
Comment (kamiyaa) [inline src/commands/file_ops.rs:87]: A bit confused on what this is trying to do:
let's say we have `/home/user/file.txt`, would this copy `/home/user` into the clipboard?
If so, you can use:
```rust
context
    .tab_context_ref()
    .curr_tab_ref()
    .curr_list_ref().file_path();
```

**Review by kamiyaa — CHANGES_REQUESTED** [review]
Good work so far. I've requested some changes.
In addition, please run the code through `cargo fmt`

On another note, I will need to add CI/CD to dev branch.

File: src/commands/file_ops.rs (hunk 1 above)
Comment (makeefu) [reply src/commands/file_ops.rs]: Done!

File: src/commands/file_ops.rs (hunk 2 above)
Comment (makeefu) [reply src/commands/file_ops.rs]: Thank you! This is really simpler!

**Comment by kamiyaa** [comment]
LGTM!

**Pull request merged**